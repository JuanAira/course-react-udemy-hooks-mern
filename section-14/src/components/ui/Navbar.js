import React from 'react'
import { Link, NavLink, useNavigate } from 'react-router-dom'

export const Navbar = () => {
    const navigate = useNavigate();

    const handleLogout = () => {
        navigate('/login', {
            replace: true
        });
    }

    const parseClassName = ({ isActive }) => {
        return isActive ? 'nav-item nav-link active' : 'nav-item nav-link';
    };

    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <Link
                className="navbar-brand"
                to="/"
            >
                Asociaciones
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">
                    <NavLink
                        className={({ isActive }) => parseClassName({ isActive})}
                        to="/marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink
                        className={({ isActive }) => parseClassName({ isActive })}
                        to="/dc"
                    >
                        DC
                    </NavLink>

                    <NavLink
                        className={({ isActive }) => parseClassName({ isActive })}
                        to="/search"
                    >
                        Search
                    </NavLink>
                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
                <ul className="navbar-nav ml-auto">
                    <span className="nav-item nav-link text-info">
                        Juanga
                    </span>

                    <button
                        className="nav-item nav-link btn"
                        onClick={handleLogout}
                    >
                        Logout
                    </button>
                </ul>
            </div>
        </nav>
    );
};