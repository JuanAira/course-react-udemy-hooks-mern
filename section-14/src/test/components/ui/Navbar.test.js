import { MemoryRouter, Routes, Route } from 'react-router-dom'
import { mount } from 'enzyme';

import { Navbar } from '../../../components/ui/Navbar';
import { AuthContext } from '../../../auth/authContext';
import { types } from '../../../types/types';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}));

describe('Test in <Navbar />', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            name: 'Juanga',
            logged: true
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter initialEntries={['/']}>
                <Routes>
                    <Route path="/" element={<Navbar />} />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    test('should to match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Juanga');
    });

    test('should call logout, call navigate and dispatch with arguments', () => {
        wrapper.find('button').prop('onClick')();

        expect(contextValue.dispatch).toHaveBeenCalledWith({ 'type': types.logout });
        expect(mockNavigate).toHaveBeenCalledWith('/login', { replace: true });
    });
});
