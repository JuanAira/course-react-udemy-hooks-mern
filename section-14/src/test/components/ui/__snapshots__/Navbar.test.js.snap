// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Navbar /> should to match snapshot 1`] = `ReactWrapper {}`;
